lib/matplotlib/testing/compare.py

```diff
@@ -154,15 +154,16 @@ def encode_and_escape(name):
             + b") run flush\n")
         self._proc.stdin.flush()
         # GS> if nothing left on the stack; GS<n> if n items left on the stack.
-        err = self._read_until(b"GS")
-        stack = self._read_until(b">")
+        err = self._read_until((b"GS<", b"GS>"))
+        stack = ""
+        if err.endswith(b"GS<"):
+            stack = self._read_until(b">")
         if stack or not os.path.exists(dest):
-            stack_size = int(stack[1:]) if stack else 0
+            stack_size = int(stack[1:-1]) if stack else 0
```

Comment:
I think it's now `stack[:-1]`?  Previously the `[1:]` was needed to strip out the `<`, but not anymore (I think?).

Reply:
Indeed.